Sources/Basics/NativePathExtensions.swift

```diff
@@ -0,0 +1,32 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import struct TSCBasic.AbsolutePath
+
+extension String {
+    public func _nativePathString(escaped: Bool) -> Self {
```

Comment:
do we really need this as String API, or can we do this as an AbsolutePath API only? if some of the call sites are strings, I prefer we convert them to absolute path (which should be more type safe anyways) then introduce a public extension on String for something that is clearly path specific

Reply:
I can try to do the conversions if you really want.